src/Emulator/Cores/Arm-M/CortexM.cs

```diff
@@ -204,7 +205,7 @@ private void OnPRIMASKWrite(int value)
         [Export]
         private int PendingMaskedIRQ()
         {
-            return nvic.MaskedInterruptPresent.WaitOne(0) ? 1 : 0;
+            return nvic.MaskedInterruptPresent ? 1 : 0;
```

Comment:
This is a little bit controversial - I need to trace down why was it here in the first place. It looks ok now, but just want to be 100% sure

Reply:
Agreed this definitely looks like a substantial/risky change. I really couldn't get my head around why this might have been here in the first place so unfortunately, I won't be of much help :) What I do know is that the overhead of the ManualResetEvent is not negligible.

Reply:
Ok, we believe it's good to be moved to `bool`. `MRE` was used back in the days when timers had their own threads, but this is no longer the case!